test/functional/tests/lazy_writes/cleaning_policy/test_switch_cleaning_policy.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright(c) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+from datetime import timedelta
+from time import sleep
+
+import pytest
+from api.cas import casadm
+from api.cas.cache_config import CacheMode, CleaningPolicy
+from core.test_run_utils import TestRun
+from storage_devices.disk import DiskTypeSet, DiskType, DiskTypeLowerThan
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import ReadWrite, IoEngine
+from test_utils.size import Size, Unit
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_switch_cleaning_policy():
+    """
+        title: Dynamic Cleaning Policy Switching - IO reads/writes.
+        description: Verify that cleaning policy switching works properly
+                     during working IO on CAS device.
+        pass_criteria:
+          - cleaning policy switching properly
+    """
+    with TestRun.step("Prepare devices."):
+        cache_disk = TestRun.disks['cache']
+        cache_disk.create_partitions([Size(100, Unit.MebiByte)])
+        cache_dev = cache_disk.partitions[0]
+
+        core_disk = TestRun.disks['core']
+        core_disk.create_partitions([Size(400, Unit.MebiByte)])
+        core_dev = core_disk.partitions[0]
+
+    with TestRun.step("Start cache in Write-Back mode and add core."):
+        cache = casadm.start_cache(cache_dev, cache_mode=CacheMode.WB)
+        cache.set_cleaning_policy(CleaningPolicy.alru)
+        core = cache.add_core(core_dev)
+
+    with TestRun.step("Start fio on OpenCAS device."):
+        fio = (Fio().create_command()
+               .time_based()
+               .run_time(timedelta(minutes=15))
+               .read_write(ReadWrite.readwrite)
+               .block_size(Size(1, Unit.Blocks4096))
+               .direct()
+               .io_engine(IoEngine.libaio)
+               .target(core.path))
+        fio_pid = fio.run_in_background()
+
+    while TestRun.executor.check_if_process_exists(fio_pid):
```

Comment:
You can use following construction to make sure that fio ended with 0 exit code after changing cleaning policy (see usage in [test/functional/tests/stress/test_stress_change_io_class_config_io.py](https://github.com/Open-CAS/open-cas-linux/pull/1190/files#diff-ea8a1475917ea88a532b9341ba97685b6ac0d7152bb48c7818a364098a7673c1) ):

fio_task = start_async_func(run_io, cores)

 while fio_task.done() is False:
 (...)

fio_result = fio_task.result()
if fio_result.exit_code != 0:
    TestRun.fail("Fio ended with an error!")